sklearn/model_selection/_classification_threshold.py

```diff
@@ -931,6 +932,19 @@ def _fit(self, X, y, **params):
         best_idx = objective_scores.argmax()
         self.best_score_ = objective_scores[best_idx]
         self.best_threshold_ = decision_thresholds[best_idx]
+        if (
+            objective_scores.max() - objective_scores.min()
+            <= np.finfo(objective_scores.dtype).eps
```

Comment:
I wonder if this is something we should be doing everywhere where we "search" over a few models using a scorer. Kind of arbitrary to have it here and not in other places.

Reply:
I would also be in favor of raising a similar warning for `*SearchCV` meta-estimators when `mean_test_score` is constant for all hyper-parameters. Not sure how frequent this is though.

But it's true that it's actually not that frequent for `TunedThresholdClassifierCV` either. 

The original problem I encountered that triggered the selection of an extreme threshold is actually of a different nature. Let me update this PR accordingly to discuss that further (maybe tomorrow).

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/29082#issuecomment-2129083636.

Reply:
I think that checking for the constant case is not really important but I do not expect this to happen often in practice. The other cases (extreme thresholds due to lack of trade-off) are more useful to warn against (and they do easily happen in practice as shown in the tests).

But I would rather keep the constant warning to make the warning message more precise for this particular edge case.